streams/src/main/java/org/apache/kafka/streams/kstream/Predicate.java

```diff
@@ -0,0 +1,24 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.kafka.streams.kstream;
+
+public interface Predicate<K, V> {
+
+    boolean apply(K key, V value);
```

Comment:
Renaming `apply` to `test` now gives us a slow, non-breaking migration path so that when moving to Java 8 we could:
* change `Predicate<K,V>` to extend `java.util.function.BiPredicate<K,V>` and remove the `apply` method from `Predicate<K,V>`;
* change `KStream` methods to use `BiPredicate<K,V>` instead of this `Predicate<K,V>`;
* deprecate `Predicate<K,V>` and tell people to use `BiPredicate` instead.

No users would have to change their code, although anyone using `Predicate<K,V>` directly would get deprecation warnings and could but would not be required to change. In a later release, we can remove `Predicate<K,V>`, and anyone *still* using it would then have to change their code.

OTOH, if we keep `apply` as it is now, then when we moved to Java 8 and change to  `java.util.function.BiPredicate<K,V>`, all users directly using the interface would *have* to change.

Of course, neither of these options affects those users who are already on Java 8 and using lambdas rather than `Predicate<K,V>` implementation classes.